### PR TITLE
[UI-side compositing] Have pinch zoom run "snap-back" animation in commitTransientZoom

### PR DESCRIPTION
**Pull request by nmoucht**
#### 47115d5fb5b3ffa21f51b31049870299ff5d194c
<pre>
[UI-side compositing] Have pinch zoom run &quot;snap-back&quot; animation in commitTransientZoom
<a href="https://bugs.webkit.org/show_bug.cgi?id=251968">https://bugs.webkit.org/show_bug.cgi?id=251968</a>
rdar://105203991

Reviewed by Simon Fraser.

When the user has pinch zoomed beyond the minimum or maximum scale allowed, we
need to perform a snap-back animation, like we do in non ui-side compositing
mode. To do this we create a CAAnimation with a transform to the clipped scale
and origin. We also add the WebPage commitTransientZoom IPC to a completion block
to be run after the animation finishes. We still need to figure out how to properly
constrain the origin as we do in TiledCoreAnimationDrawingArea. This will need to
involve some way of getting web process state into the UI process, including the
visibleContentRect and several different FrameView members.

* Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeDrawingAreaProxyMac.mm:
(WebKit::transientZoomSnapAnimationForKeyPath):
(WebKit::RemoteLayerTreeDrawingAreaProxyMac::commitTransientZoom):
* Source/WebKit/WebProcess/WebPage/mac/TiledCoreAnimationDrawingArea.mm:
(WebKit::TiledCoreAnimationDrawingArea::commitTransientZoom):

Canonical link: <a href="https://commits.webkit.org/260161@main">https://commits.webkit.org/260161@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/0ca9db639b1c8d6f171d86435012f60e5c1bb677

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/107244 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/16288 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/40187 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/116402 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/115863 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/17737 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/7558 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/99402 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/113009 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/13388 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/96570 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/41004 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/95289 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/28203 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/82773 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/9349 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/29549 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/9987 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/6446 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/15504 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49129 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/11519 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3805 "Built successfully and passed tests") | | | |

### CODE REVIEW

File: Source/WebKit/WebProcess/WebPage/DrawingAreaCocoa.mm [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/*
```
Comment (smfr) [inline Source/WebKit/WebProcess/WebPage/DrawingAreaCocoa.mm]: This file should go into `WebPage/cocoa`

File: Source/WebKit/WebProcess/WebPage/DrawingAreaCocoa.mm [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+
+#if PLATFORM(COCOA)
+#include "DrawingArea.h"
+
+#include "DrawingAreaMessages.h"
```
Comment (smfr) [inline Source/WebKit/WebProcess/WebPage/DrawingAreaCocoa.mm]: Is this necessary?

File: Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm [hunk 3]
```diff
@@ -170,6 +170,7 @@
 #define WEBPAGE_RELEASE_LOG_ERROR(channel, fmt, ...) RELEASE_LOG_ERROR(channel, "%p - WebPage::" fmt, this, ##__VA_ARGS__)
 
 namespace WebKit {
+using namespace WebCore;
```
Comment (smfr) [inline Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:173]: Why this change?

**Review by smfr — APPROVED** [review]

File: Source/WebKit/WebProcess/WebPage/DrawingAreaCocoa.mm (hunk 1 above)
Comment (nmoucht) [reply Source/WebKit/WebProcess/WebPage/DrawingAreaCocoa.mm]: Fixed.

File: Source/WebKit/WebProcess/WebPage/DrawingAreaCocoa.mm (hunk 2 above)
Comment (nmoucht) [reply Source/WebKit/WebProcess/WebPage/DrawingAreaCocoa.mm]: Removed.

File: Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm (hunk 3 above)
Comment (nmoucht) [reply Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:173]: This was to fix the build errors on iOS.
